sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/operator/BeamSqlPrimitive.java

```diff
@@ -131,6 +131,8 @@ public boolean accept() {
       case CHAR:
       case VARCHAR:
         return value instanceof String || value instanceof NlsString;
+      case VARBINARY:
+        return value instanceof byte[];
```

Comment:
Calcite is using `ByteString` for both BINARY and VARBINARY: https://calcite.apache.org/avatica/apidocs/org/apache/calcite/avatica/util/ByteString.html We should match, as that will minimize pain when I pull in the Calcite Rex code generation.

Reply:
good point, I only considered the type in `Row`, let me update it.